app/controllers/ArticlesController.scala

```diff
@@ -20,9 +20,9 @@ class ArticlesController @Inject() extends Controller {
     val optTitle = request.getQueryString("title")
 
     val filteredArticles = Articles.getArticles
-      .filter( article => optAuthor.fold(true)(author => author == article.author))
-      .filter( article => optContent.fold(true)(content => content == article.author))
-      .filter( article => optTitle.fold(true)(title => title == article.author))
+      .filter( article => optAuthor.fold(true)(author => article.author.toLowerCase contains author.toLowerCase))
```

Comment:
This lambda can be generalized by this for example:
```scala
def contains(str1: String)(str2: String) = str1.toLowerCase contains str2.toLowerCase
```
This way you can do `.filter( article => optAuthor.fold(true)(contains(article.author)))`